colcon_clean/base_handler/install.py

```diff
@@ -24,8 +29,32 @@ def add_arguments(self, *, parser):  # noqa: D102
             help='The base path for all install directories '
                  f'(default: {self.base_path})')
 
-    def get_workspace_paths(self, *, args):  # noqa: D102
-        return [args.install_base]
-
-    def get_package_paths(self, *, args, pkg):  # noqa: D102
-        return [os.path.join(args.install_base, pkg.name)]
+    def get_workspace_paths(self, *, args) -> List[Path]:  # noqa: D102
+        return [Path(args.install_base)]
+
+    def get_package_paths(  # noqa: D102
+        self, *, args, pkg: PackageDescriptor
+    ) -> List[Path]:
+        paths: List[Path] = [
+            Path(args.install_base) / pkg.name,
+            Path(args.install_base) / 'share' / pkg.name,
+        ]
+        if hasattr(args, 'build_base'):
+            manifest_path = (
+                Path(args.build_base) / pkg.name / 'install_manifest.txt'
+            )
+            if not manifest_path.is_file():
+                logger.debug(
+                    f'No {manifest_path.name} found for {pkg.name} at '
+                    f"'{manifest_path}'"
+                )
+                return paths
+            with manifest_path.open() as f:
```

Comment:
Could we add a test using such manifest files, to avoid ever breaking it in the future? Perhaps just extending on of the example packages in https://github.com/colcon/colcon-clean/tree/master/test/resources/test_src/test-repo ?